### PR TITLE
Lint yaml files, add linter to pre-commit tool

### LINKED ISSUE #199: Create a yaml format checker as a pre-commit hook

### PR DESCRIPTION
**Pull request by ng390**
Lint yaml files, add linter to pre-commit tool. Closes #199.

### CODE REVIEW

**Comment by davidslater** [comment]
Need to add yamllint to `test-requirements.txt`

**Comment by ng390** [comment]
> Need to add yamllint to `test-requirements.txt`

Good catch -- I also should have added a yamlllint check to the CI testing.

File: .github/workflows/ci_test.yml [hunk 1]
```diff
@@ -48,11 +50,23 @@ jobs:
...
   json-format-test:
     name: JSON file format
     runs-on: ubuntu-18.04
     steps:
       - uses: actions/checkout@v1
       - uses: actions/setup-python@v1
-      - name: Ensure contributor used ("python -m tools.format_json") before commit
+      - name: >
+          Ensure contributor used ("python -m tools.format_json")
+          before commit
```
Comment (davidslater) [inline .github/workflows/ci_test.yml]: Does this need to be broken?

File: .github/workflows/ci_test.yml [hunk 2]
```diff
@@ -1,5 +1,7 @@
+---
 name: GitHub CI
 
+# yamllint disable-line rule:truthy
```
Comment (davidslater) [inline .github/workflows/ci_test.yml]: Let's disable truthy in the `.yamllint` file - I would prefer not having to add yamllint comments in our yaml files.

File: .yamllint [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+extends: default
+
+rules:
```
Comment (davidslater) [inline .yamllint]: I would recommend adding additional constraints that will enforce white spaces to be consistent. Example: "spaces: 2" https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.indentation

Ideally, this linting will both catch syntax errors and minimize diffs from PRs.

File: .github/workflows/ci_test.yml (hunk 1 above)
Comment (ng390) [reply .github/workflows/ci_test.yml]: Good catch, will revert.

**Comment by davidslater** [comment]
I'd like to get @seanpmorgan 's thoughts on this as well, since this will impact all yaml files in the repo.

**Comment by ng390** [comment]
> Can you also add a few lines to STYLE.md under Formatting?

Updated STYLE.md.

**Pull request merged**